src/LanguageServer/Impl/Implementation/IndexingFileParser.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Microsoft.PythonTools.Analysis.Indexing;
+using Microsoft.PythonTools.Parsing;
+
+namespace Microsoft.Python.LanguageServer.Implementation {
+    class IndexingFileParser {
+        private SymbolIndex _index;
+        private PythonLanguageVersion _pythonLanguageVersion;
+
+        public IndexingFileParser(SymbolIndex index, PythonLanguageVersion pythonLanguageVersion) {
+            _index = index;
+            _pythonLanguageVersion = pythonLanguageVersion;
+        }
+
+        public void ParseForIndex(string path) {
+            var uri = new Uri(path);
+            using (var stream = new StreamReader(uri.AbsolutePath)) {
```

Comment:
`StringReader`

Reply:
He'd have to change the interface, since that's a path to a file, not the file contents.